### PR TITLE
Fix build issues on Windows

### PR DESCRIPTION
**Pull request by yanang007**
This pull request mainly fixes issues on building and installing under Windows.

Changes
---------
1. Fixes `runtime_library_dirs` unsupported on Windows (#16, #17).
2. Makes it possible to build a binary distribution.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -8,23 +9,54 @@
...
+if 'install' in sys.argv[1:]:
+    if sys.platform == "win32":
+        data_files = [('', [lib_file_path])]
+    else:
+        runtime_lib_dirs = [numpy.get_include(),] + AMGX_lib_dirs
+else:
+    # bdist_wheel or other subcommand
+    data_files = [(
+        os.path.relpath(get_python_lib(), sys.prefix), # should put the runtime under "lib\site-packages"
+        [lib_file_path]
```
Comment (shwina) [inline setup.py:57]: 1. Is this essentially packaging the AMGX library into the wheel? I'm not sure about what issues this could cause on Windows, but on Linux, I'd be concerned about things like glibc binary compatibility or CUDA version compatibility. Perhaps, can we break this off into a separate PR and discuss there?

2. What about other `setup.py` commands like `python setup.py develop`?

**Comment by shwina** [comment]
Thanks, @yanang007! Can you confirm if you need anything further for Windows or if things work with this PR? I'll verify Linux builds tomorrow and merge.

I wish we had CI set up for this repo but I don't have the bandwidth for that at the moment..

**Comment by yanang007** [comment]
Yes, it has enabled at least `python setup.py install` on Windows. And I had also pushed another commit to leave a warning message with final solution for ImportError on Windows.

As for other subcommands like "bdist_wheel", I think it should be discussed in another packaging-related PR.

**Comment by yanang007** [comment]
> Thanks, @yanang007! Can you confirm if you need anything further for Windows or if things work with this PR? I'll verify Linux builds tomorrow and merge.
> 
> I wish we had CI set up for this repo but I don't have the bandwidth for that at the moment..

I actually am trying to set up a GitHub CI to make a distributable package with AMGX library shipped together. And I found another compatibility issue with Python 3.10+, where Python complains about the mismatching signature of `c_print_callback` and `AMGX_print_callback`.
https://github.com/shwina/pyamgx/blob/6c4331e0bfe8c4c5e1d6aa54147db684a90179f5/pyamgx/System.pyx#L52

This can be fixed by simply adding a noexcept.
```python
cdef void c_print_callback(char *msg, int length) noexcept:
```

You probably want to test this fix as well.

**Review by shwina — APPROVED** [review]
Thanks!
